### DISCUSSION TITLE
Flashing Shelly 2.5 to Mongoose OS via USB-COM (Currently HAA Installed)

### DISCUSSION THREAD

**Discussion by genius23pl** [mongoose-os-apps/shelly-homekit#1532] [General]
Hi everyone,

I need help with flashing my Shelly 2.5 to Mongoose OS using a USB-to-Serial (USB-COM) adapter. Currently, my device is running HAA firmware, and I don't have the original Shelly firmware to revert to.

**Comment by markirb** [comment]
What's the question?
Cabling?
If you want to flash just use mos-tool with the downloadable zip files.

**Reply by genius23pl** [reply]
I was just getting to that, but I did it differently - I had one shelly 2.5 with the original shelly software - I made a backup of it, uploaded this backup to all the shelly 2.5 drivers , and then did the upgrade to mongoose-os via OTA.
